src/plugins/reporter-mailx.c

```diff
@@ -99,22 +117,52 @@ static void create_and_send_email(
     unsigned arg_size = 0;
     args = append_str_to_vector(args, &arg_size, "/bin/mailx");
 
-    char *dsc = make_description_mailx(problem_data, CD_TEXT_ATT_SIZE_LOGGER);
+    problem_formatter_t *pf = problem_formatter_new();
+    /* formatting file is not set */
+    if (fmt_file == NULL)
+    {
+        env = getenv("Mailx_Subject");
+        const char *subject = (env ? env : get_map_string_item_or_NULL(settings, "Subject") ? : PR_DEFAULT_SUBJECT);
+
+        char *format_string = xasprintf(PR_MAILX_TEMPLATE, subject);
+
+        /* attaching binary file to the email */
+        if (send_binary_data)
+            format_string = append_to_malloced_string(format_string, PR_ATTACH_BINARY);
+
+        if (problem_formatter_load_string(pf, format_string))
+            error_msg_and_die("Invalid problem report format string");
```

Comment:
I would explicitly mention which string is invalid ("defautl") and also that it is a bug in code.